### PR TITLE
Porting multi-z support updates.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Porting /tg/station PRs #40428, #40558, #41902, #41967, #42123, #42716, #45437, #45531, #47046, #47097, #47336, #47500, #47501, #47557, #47588. again with the power of copy and paste (it helps me review the code step by step honk).

## Why It's Good For The Game
Updating multi-z support. 
Most of the aforementioned PRs simply concern fixes and tweaks of a very unimplemented feature so no wall of text changelog.

## Changelog
:cl:  Kevinz000, Cruix, MrStonedOne, Denton, Kmc2000, Anturk, MrDoomBringer, Dennok, TheChosenEvilOne, Ghommie
add: Added support for Multi-Z power, atmospherics and disposals
fix: massive service department nerf: space can no longer be extra crispy.
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/environmental/LINDA_system.dm [hunk 1]
```diff
@@ -8,42 +10,47 @@
...
 /atom/movable/proc/BlockSuperconductivity() // objects that block air and don't let superconductivity act. Only firelocks atm.
-	return 0
+	return FALSE
 
 /turf/proc/CalculateAdjacentTurfs()
-	var/list/atmos_adjacent_turfs = src.atmos_adjacent_turfs
-	for(var/direction in GLOB.cardinals)
-		var/turf/T = get_step(src, direction)
-		if(!T)
+	var/canpass = CANATMOSPASS(src, src)
```
Comment (deathride58) [inline code/modules/atmospherics/environmental/LINDA_system.dm:47]: These two vars are only used once in this proc. You can safely squash them into the places where they're used.

**Review by deathride58 — DISMISSED** [review]
Calculateadjacentturfs() could probably be turned into a define in the interest of avoiding proc call overhead, but that's outside of this PR's scope. Doing some local tinkering, the overhead this PR adds to atmos is very much worth the functionality.

File: code/modules/atmospherics/environmental/LINDA_system.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/atmospherics/environmental/LINDA_system.dm:47]: These two variables are actually encapsulated in a `for(var/direction in GLOB.cardinals_multiz)` loop. Doing what you suggest here would be an obvious performance drawback.

**Comment by Ghommie** [comment]
By the by I'm also porting the atmos adjacency subsystem from /tg/station (PR #43447 by AnturK) since I'm already touching atmospheric. Cool with that?

File: code/modules/atmospherics/environmental/LINDA_system.dm (hunk 1 above)
Comment (deathride58) [reply code/modules/atmospherics/environmental/LINDA_system.dm:47]: ohhhhhhhh. yeah im dum this is good

**Review by silicons — APPROVED** [review]

**Review by silicons — APPROVED** [review]

**Pull request merged**